### PR TITLE
2048: Miscellaneous improvements

### PR DESCRIPTION
**Pull request by dykatz**
This PR makes a few improvements to 2048.

1. There was a minor bug in the settings menu where opening it would load the target tile value as one power-of-2 larger than it should have been. This would result in the value increasing every time that the settings were opened and re-saved.
2. The dialog boxes for the win condition were a little bit janky. The player would be asked if they'd like to continue endlessly, and then only if they did not want to, they would get a second dialog with game statistics. I consolidated them into a single dialog.
3. The "New Game" and "Settings" actions were missing icons.

### CODE REVIEW

File: Userland/Games/2048/main.cpp [hunk 1]
```diff
@@ -190,7 +186,7 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     })));
 
     TRY(game_menu->try_add_separator());
-    TRY(game_menu->try_add_action(GUI::Action::create("&Settings...", [&](auto&) {
+    TRY(game_menu->try_add_action(GUI::Action::create("&Settings...", Gfx::Bitmap::try_load_from_file("/res/icons/16x16/settings.png").release_value_but_fixme_should_propagate_errors(), [&](auto&) {
```
Comment (bgianfo) [inline Userland/Games/2048/main.cpp:189]: Let’s not add more error propagation debt.
```suggestion
    TRY(game_menu->try_add_action(GUI::Action::create("&Settings...", TRY(Gfx::Bitmap::try_load_from_file("/res/icons/16x16/settings.png")), [&](auto&) {
```

We should apply the same fix above as well,

**Review by bgianfo — CHANGES_REQUESTED** [review]

**Comment by bgianfo** [comment]
I didn’t look closely but is the off by one related to #11400?

**Review by bgianfo — APPROVED** [review]

**Comment by dykatz** [comment]
> I didn’t look closely but is the off by one fix related to #11400?

I don't think so. I took a look at that to see if there was an issue filed for this behavior when I was making the PR. That appears to be something about the evil difficulty.

**Comment by mitchell486** [comment]
@dykatz Good afternoon! I was in the process of submitting PR #13261, which actually reversing the first commit change of this PR. After starting to create that PR, I decided to search a bit further and I found this one which I was obviously unaware of. Currently, I can reproduce the exact opposite behavior you had fixed in the commit `fabac72`. I actually mentioned the newly decrementing `Target tile` problem in Issue #11400 the other day, because I thought they were related. (_I realize now that I am wrong, and those comments actually belong in this thread, or an entirely separate issue._)

I sincerely apologize about any confusion so far, but as of testing earlier today the following steps decrement the `Target tile` setting automatically:

* Open the 2048 game
* Open the settings menu.
* Manually set the `Target tile` value to 11 (or any value)
* Click Ok (and Ok again on the second prompt about needing to start a new game)
* Open the settings menu a 2nd time
* NOTE: The `Target tile` value has now decremented from 11 to 10. It will continue to decrement until it gets to 0, at which point, I believe the code base will force it to statically always be set to 2048 and never need to reference the `Target tile` setting again, because it's at 0.


I am not a C++ programming be default, so I will defer to anyone that knows more about this issue. However, I suspected that the confusion and/or bug might have happened from interchanged usage of the words "target_tile" and "target". (_Although, it could be something else entirely._)

Please let me know if I should close my current PR (which needs to be amended because I didn't read all of the rules before trying to help. 😞). For now, I'm going to at least amend my current commit message that was incorrectly done, and leave the PR as is until there is more discussion from either this thread, or that one. Thank you.

**Pull request merged**